pandas/tests/sparse/test_series.py

```diff
@@ -88,6 +88,29 @@ def setup_method(self, method):
         self.ziseries2 = SparseSeries(arr, index=index, kind='integer',
                                       fill_value=0)
 
+    def test_constructor_data_input(self):
+        constructor_dict = {1: 1.}
+        index = [0, 1, 2]
+        series = pd.Series(constructor_dict)
+
+        expected = SparseSeries(series, index=index)
+
+        result = SparseSeries(constructor_dict, index=index)
+        tm.assert_sp_series_equal(result, expected)
+
+        # Series and dictionary with no index: see gh-16905
+        expected = SparseSeries(series)
+
+        result = SparseSeries(constructor_dict)
+        tm.assert_sp_series_equal(result, expected)
+
+        # Series with index and dictionary with no index
```

Comment:
I was testing input with a series, but that is test turned out to be redundant. I removed it from the most recent commit.